frontend/components/ManageAllowList.tsx

```diff
@@ -96,77 +111,157 @@ export default function ManageAllowList() {
+        <Text fontFamily="Bungee" fontWeight={400} fontSize={'14px'}>
+          Checking status...
+        </Text>
+      );
+    }
+    if (isAddingToAllowList) {
+      return (
+        <Text fontFamily="Bungee" fontWeight={400} fontSize={'14px'}>
+          Adding to allow list...
+        </Text>
+      );
+    }
+    if (isRemovingFromAllowList) {
+      return (
+        <Text fontFamily="Bungee" fontWeight={400} fontSize={'14px'}>
+          Removing from allow list...
+        </Text>
+      );
+    }
+    // When there is a token check message, show it with the CheckCircleIcon
+    if (tokenCheckMessage) {
+      return (
+        <Flex alignItems="center">
+          <Text fontFamily="Bungee" fontWeight={400} fontSize={'14px'}>
+            {tokenCheckMessage + ' '}
+            <span
+              style={{
+                display: 'inline-flex',
+                alignItems: 'center',
+                position: 'relative',
+                top: '1px',
+              }}
+            >
+              <BiSolidCheckCircle />
+            </span>
           </Text>
-          <FormControl>
-            <FormLabel>Token Address</FormLabel>
-            <Input value={tokenAddress} onChange={handleChange} />
-          </FormControl>
-          <Text mt={4} minH={6}>
-            {actionText}
+        </Flex>
+      );
+    }
+    return null; // Return null if there's no specific message to display
+  };
+
+  return (
+    <Box>
+      <Text
+        fontSize="20px"
+        fontWeight="bold"
+        fontFamily="Bungee"
+        color="dark.purple.400"
+      >
+        Manage Assets Allowlist
+      </Text>
+      <Text
+        mb={4}
+        mt={4}
+        fontWeight={600}
+        fontFamily="Montserrat"
+        textAlign="start"
+      >
+        If you want to mint, receive, or swap certain LSP7 or LSP8 tokens you
+        can add them to the allowlist beforehand so that they don’t get
+        automatically redirected to your Grave.
+      </Text>
+      <FormControl textAlign="start">
+        <FormLabel fontFamily="Bungee" fontWeight={400} fontSize={'14px'}>
+          ASSET ADDRESS
+        </FormLabel>
+        <Box display="flex" alignItems="center" h="50px">
+          <Input
+            width="314px"
+            height="25px"
+            fontFamily={'Bungee'}
+            backgroundColor="white"
+            borderColor={'var(--chakra-colors-dark-purple-500)'}
+            _hover={{
+              borderColor: 'var(--chakra-colors-dark-purple-500)',
+            }}
+            _focus={{
+              borderColor: 'var(--chakra-colors-dark-purple-500)',
+              boxShadow: 'none',
+            }}
+            value={tokenAddress}
+            onChange={handleChange}
+            placeholder="PASTE ASSET ADDRESS"
+            _placeholder={{
+              fontWeight: 'bold',
+              color: 'var(--chakra-colors-dark-purple-200)',
+            }}
+            onBlur={fetchTokenAllowListStatus}
```

Comment:
rather than fetching onBlur think it should be more specific, because it's funky that user has to click outside of field for it to process. onChange would be better imo @Ledespana

Reply:
Onchange even with a debouncing effect will make several calls, showing the erros each time